s3/s3.go

```diff
@@ -1035,28 +1065,7 @@ func (s3 *S3) setupHttpRequest(req *request) (*http.Request, error) {
 // If resp is not nil, the XML data contained in the response
 // body will be unmarshalled on it.
 func (s3 *S3) doHttpRequest(hreq *http.Request, resp interface{}) (*http.Response, error) {
-	c := http.Client{
```

Comment:
hehehe i was surprised too... that means they never persist connection. I am not sure what their intention is. Maybe they don't have an use case that they need to persist the connection. They'd rather close the conns on each use.